src/tekton/save.ts

```diff
@@ -0,0 +1,70 @@
+/*-----------------------------------------------------------------------------------------------
+ *  Copyright (c) Red Hat, Inc. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE file in the project root for license information.
+ *-----------------------------------------------------------------------------------------------*/
+
+import * as os from 'os';
+import * as path from 'path';
+import { cli } from '../cli';
+import * as fs from 'fs-extra';
+import * as yaml from 'js-yaml';
+import * as vscode from 'vscode';
+import { contextGlobalState } from '../extension';
+import { tektonYaml } from '../yaml-support/tkn-yaml';
+import { pipelineExplorer } from '../pipeline/pipelineExplorer';
+import { getStderrString, Command, newK8sCommand } from '../tkn';
+
+
+function checkSave(): boolean {
+  return vscode.workspace
+    .getConfiguration('vs-tekton')
+    .get<boolean>('save');
+}
+
+export async function save(document: vscode.TextDocument): Promise<void> {
```

Comment:
name of this function may be `onSave` or `updateTektonResource` as you not perform save, you just handle `save` event